### ISSUE TITLE
BaseNodeStateMachine should not be dependent on Peer Manager

### ISSUE THREAD

**Issue by CjS77** [tari-project/tari#1581]
..or any other heavyweight networking or comms-stack code.

The banning logic added in #1578 is a crucial step in preventing malicious or buggy nodes from preventing overall network health and chain synching.

However, #1578 tightly couples the peer and connection managers to the base node state machine which makes testing difficult and breaks the Single Responsibility principle. 

In actual fact, the inclusion of `OutboundNodeCommsInterface` is already problematic for the exact same reasons and should also be decoupled from the state machine.

Long term, the three components should be decoupled using one of these approaches:

## Coding to interfaces / traits

The functionality that the BNSM is currently making use of (e.g. determining whether a given peer is banned or not) can be abstracted away behind a trait. Tests can implement mock implementations of these traits to break coupling and keep tests small and focussed.

The drawback to this approach is that currently, `async` functions are not allowed in traits; there are workarounds; but they do complicate the implementation.

Another possible drawback: Do the traits solve the Single-Responsibility violations we currently experience?

## Using streams and channels

Implement mpsc, pubsub or broadcast channels to send/receive state and event information to and from the BNSM rather than embedding heavyweight components into the BNSM itself.

This is a highly decoupled approach, but one that will result in a much higher net gain in lines of code than the Trait approach.

Testing is simplified by hand-crafting messages to push down the tx side of channels; and listening for specific messages on the rx side.

We have a lot of tools available already to implement this approach (async::mpsc, tari_broadcast, pub_sub).

This approach is better suited to "fire-and-forget" strategies by and large vs Request-Response; although there are ways around this too.

tl;dr Which is the best approach? It depends. Whats clear is that tight coupling is a PITA and not the way to go. This issue is here to get some discussion going on the best way forward for this specific technical debt problem.

**Comment by SWvheerden** [comment]
The whole stack is currently a mess and needs a massive refactor. 
Not going to get to this soon, going to close this for now.
